presto-spark-base/src/main/java/com/facebook/presto/spark/execution/PrestoSparkTaskExecutorFactory.java

```diff
@@ -744,7 +743,7 @@ private static TaskInfo createTaskInfo(
         {
             TaskId taskId = taskContext.getTaskId();
             TaskState taskState = taskContext.getState();
-            TaskStats taskStats = taskContext.getTaskStats();
+            TaskStats taskStats = taskContext.getTaskStats().summarizeFinal();
```

Comment:
is this needed ? TaskInfo.summarize() will handle this.

Reply:
It's pretty much the same, `TaskInfo.summarize` calls `TaskStats.summarizeFinal`. Just trying to save some object allocations =D

Reply:
It will create new objects for every call of `summarizeFinal`.